### PR TITLE
Add haproxy stretch image

### PR DESCRIPTION
**Pull request by diptadas**
#997

### CODE REVIEW

**Comment by george-oakling** [comment]
Why not alpine?

**Comment by tamalsaha** [comment]
Alpine will be the default image. But we want to give users an options , in case they hit any of the DNS quirks of alpine. https://github.com/gliderlabs/docker-alpine/blob/master/docs/caveats.md#dns

File: hack/docker/haproxy/1.8.8/setup.sh [hunk 1]
```diff
@@ -14,11 +14,13 @@ TAG=1.8.8-$TAG
 
 build() {
 	pushd $(dirname "${BASH_SOURCE}")
-	cp $REPO_ROOT/dist/voyager/voyager-alpine-amd64 voyager
+	cp $REPO_ROOT/dist/voyager/voyager-linux-amd64 voyager
 	chmod +x voyager
+	# download socklog (`socklog` not available for `stretch`, use `jessie` deb instead)
+	curl -o socklog.deb http://ftp.us.debian.org/debian/pool/main/s/socklog/socklog_2.1.0-8_amd64.deb
```
Comment (tamalsaha) [inline hack/docker/haproxy/1.8.8/setup.sh]: curl -L -o socklog.deb http://ftp.us.debian.org/debian/pool/main/s/socklog/socklog_2.1.0-8_amd64.deb

**Comment by tamalsaha** [comment]
@george-oakling , if you want to dig deep, here is a great talk from Kubeconhttps://www.youtube.com/watch?v=ZnW3k6m5AY8

**Pull request merged**